gtsam/base/std_optional_serialization.h

```diff
@@ -0,0 +1,95 @@
+// Functionality to serialize std::optional<T> to boost::archive
+// Following this:
+// PR: https://github.com/boostorg/serialization/pull/148/files#
```

Comment:
I think this should be boostorg/serialization#163, not 148 (which is for `std::variant` and a bit harder to read).

Also, I think the [boost software license](https://www.boost.org/LICENSE_1_0.txt) is supposed to be linked.

Otherwise, LGTM! :)